OpenAPITools/openapi-generator
# [csharp-netcore] HttpClient/Set TimeoutException to inner exception in case of timeout

**What is the issue**

- When a timeout occurs due to Configuration.Timeout, a TaskCanceledException is thrown, but this can not distinguish upstream whether it is a manual cancellation or a timeout.

**Correction details**

Changed to TimeoutException is thrown when a timeout occurs due to configuration.Timeout.

**How to validate the work**

- Set an extremely small value (e.g., 1 ms) in configuration.Timeout and execute the communication in production.
- At this time, check that a TimeoutException is thrown and the process is aborted.


### PR checklist
 
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)

@mandrean
@shibayan 
@Blackclaws 
@lucamazzanti
